core/src/main/java/io/aiven/klaw/service/ClusterApiService.java

```diff
@@ -1019,13 +1020,24 @@ public Map<String, Object> getConnectorDetails(
   }
 
   public ConnectorsStatus getAllKafkaConnectors(
-      String kafkaConnectHost, String protocol, String clusterIdentification, int tenantId)
+      String kafkaConnectHost,
+      String protocol,
+      String clusterIdentification,
+      int tenantId,
+      boolean getConnectorsStatuses)
       throws KlawException {
     log.info("getAllKafkaConnectors {}", kafkaConnectHost);
     getClusterApiProperties(tenantId);
     try {
       String uriGetTopics =
-          URI_GET_ALL_CONNECTORS + kafkaConnectHost + "/" + protocol + "/" + clusterIdentification;
+          URI_GET_ALL_CONNECTORS
+              + kafkaConnectHost
+              + "/"
+              + protocol
+              + "/"
+              + clusterIdentification
+              + "?connectorStatus="
```

Comment:
Static ?connectorStatus

as an aside I wonder how other projects handle these complex urls and if there is something we could learn from them?
Might be worth us checking out a few other OOS best practices for these

Reply:
agree